test/Microsoft.Azure.WebJobs.Host.EndToEndTests/InvocationFilterEndToEndTests.cs

```diff
@@ -0,0 +1,635 @@
+
+            executingContext.Logger.LogInformation("This is an authorized user!");
+            return Task.CompletedTask;
+        }
+    }
+
+    public class HTTPRequestFilter : InvocationFilterAttribute
+    {
+        public HttpRequestMessage HttpRequestToValidate { get; set; }
+
+        public override Task OnExecutingAsync(FunctionExecutingContext executingContext, CancellationToken cancellationToken)
+        {
+            executingContext.Logger.LogInformation("Test executing!");
+
+            IReadOnlyDictionary<string, object> arguments = executingContext.Arguments;
+            var testValues = string.Join(",", arguments.Values.ToArray());
+
+            // Check headers (I'm sure there's a better way to check this
+            if (testValues.Contains("Headers"))
+            {
+                executingContext.Logger.LogInformation("Found the header!");
+                // Perform a validation on the headers
+            }
+
+            return Task.CompletedTask;
+        }
+    }
+
+    public class TestFailingFilter : InvocationFilterAttribute
+    {
+        bool failPreInvocation = false;
+
+        public TestFailingFilter(bool input = false)
+        {
+            failPreInvocation = input;
+        }
+
+        public override Task OnExecutingAsync(FunctionExecutingContext executingContext, CancellationToken cancellationToken)
+        {
+            if ( failPreInvocation == true )
+            {
+                executingContext.Logger.LogInformation("Failing pre invocation!");
+                throw new Exception("Failing on purpose!");
+            }
+
+            return Task.CompletedTask;
+        }
+
+        public override Task OnExecutedAsync(FunctionExecutedContext executedContext, CancellationToken cancellationToken)
+        {
+            if (failPreInvocation == true)
+            {
+                return Task.CompletedTask;
+            }
+
+            executedContext.Logger.LogInformation("Failing post invocation!");
+            throw new Exception("Failing on purpose!");
+        }
+    }
+
+    public class TestFailingFunctionFilter : InvocationFilterAttribute
+    {
+        public override Task OnExecutedAsync(FunctionExecutedContext executedContext, CancellationToken cancellationToken)
+        {
+            executedContext.Logger.LogInformation("The function failed!");
+            executedContext.Logger.LogInformation(executedContext.Result.Exception.ToString());
+
+            return Task.CompletedTask;
+        }
+    }
+
+    internal class TestNameResolver : RandomNameResolver
+    {
+        public override string Resolve(string name)
+        {
+            if (name == "test_account")
+            {
+                return "SecondaryStorage";
+            }
+            return base.Resolve(name);
+        }
+    }
+
+    public class TestFixture : IDisposable
+    {
+        public TestFixture()
+        {
+            RandomNameResolver nameResolver = new TestNameResolver();
```

Comment:
Remove this and the reference to MultipleStorageAccountsEndToEndTests which is from a different test class.